qiskit/quantum_info/primitives/sampler/sampler.py

```diff
@@ -0,0 +1,170 @@
+        else:
+            self._circuits = [init_circuit(circuits)]
+
+        self._skip_transpilation = False
+
+    @classmethod
+    def from_backend(cls, backend: Union[Backend, BackendWrapper, Sampler]) -> "Sampler":
+        """Generates a Sampler based on a backend
+
+        Args:
+            backend: a backend, a backend wrapper, or a sampler
+
+        Returns:
+            a sampler
+        """
+        if not isinstance(backend, Sampler):
+            return cls(backend=backend)
+        return backend
+
+    @property
+    def circuits(self) -> list[QuantumCircuit]:
+        """A list of quantum circuits to be executed
+
+        Returns:
+            a list of quatum circuits
+        """
+        return self._circuits
+
+    @property
+    def preprocessed_circuits(self) -> Optional[list[QuantumCircuit]]:
+        return self._circuits
+
+    # pylint: disable=arguments-differ
+    def run(
+        self,
+        parameters: Optional[Union[list[float], list[list[float]]]] = None,
+        circuits: Optional[list[QuantumCircuit]] = None,
+        **run_options,
+    ) -> SamplerResult:
+        if circuits is not None:
+            self._circuits = circuits
+        return cast(SamplerResult, super().run(parameters, **run_options))
+
+    @property
+    def backend(self) -> Backend:
+        """
+        Returns:
+            The backend which this sampler object based on
+        """
+        return self._backend.backend
+
+    def set_skip_transpilation(self):
+        """Once the method is called, the transpilation will be skiped."""
+        self._skip_transpilation = True
+
+    def _get_quasis(
+        self, results: list[Result], num_circuits: int
+    ) -> tuple[list[QuasiDistribution], float]:
+        """Converts a list of results to quasi-probabilities and the total number of shots
+
+        Args:
+            results: a list of results
+            num_circuits: the number of circuits.
+
+        Returns:
+            a list of quasi-probabilities and the total number of shots
+
+        Raises:
+            QiskitError: if inputs are empty
+        """
+        if len(results) == 0:
+            raise QiskitError("Empty result")
+        list_counts = self._backend.get_counts(results)
+        counters: list[Counter] = [Counter() for _ in range(num_circuits)]
+        i = 0
+        for counts in list_counts:
+            for count in counts:
+                counters[i % num_circuits].update(count)
+                i += 1
+        shots = sum(counters[0].values())
+        quasis = [QuasiDistribution({k: v / shots for k, v in c.items()}) for c in counters]
+        return quasis, shots
+
+    def _postprocessing(self, result: Union[Result, BaseResult, dict]) -> SamplerResult:
+        if not isinstance(result, Result):
+            raise TypeError("result must be an instance of Result.")
+
+        raw_results = [result]
+        quasis, shots = self._get_quasis(raw_results, len(result.results))
```

Comment:
I plan to generate quasi-probabilties directly by `BaseReadoutMitigator.quasi_probabilities`